### PR TITLE
fix(medusa): category_id and q params for list products endpoint ld work

### PR DESCRIPTION
**Pull request by riqwan**
Looks like during an earlier refactor, some of the categories logic wasn't ported over to the handler that works with q params. 

what:

- adds a fix that allows queries to be made on category_id when q param is passed.

Fixes https://github.com/medusajs/medusa/issues/4885

### CODE REVIEW

File: integration-tests/api/__tests__/store/products/ff-product-categories.ts [hunk 1]
```diff
@@ -151,6 +151,27 @@ describe("/store/products", () => {
       ])
     })
 
+    it("should return a list of products queried by category_id and q", async () => {
+      const api = useApi()
+      const productName = "Test product"
+      const params = `category_id[]=${categoryWithProductId}&q=${productName}&expand=categories`
```
Comment (olivermrbl) [inline integration-tests/api/__tests__/store/products/ff-product-categories.ts]: question: Does this category contain more than one product?

**Review by olivermrbl — APPROVED** [review]
LGTM! 💪

File: integration-tests/api/__tests__/store/products/ff-product-categories.ts (hunk 1 above)
Comment (riqwan) [reply integration-tests/api/__tests__/store/products/ff-product-categories.ts]: Good point, added one with multiple products

Comment (olivermrbl) [reply integration-tests/api/__tests__/store/products/ff-product-categories.ts]: great!

**Pull request merged**